mitodl/ocw-studio
# Implemented correct UI for 'files' config items

#### Pre-Flight checklist

 
#### What are the relevant tickets?
Closes #192 

#### What's this PR do?
- Fixes the UI so "file"-type config items so they can be added/edited again
- Changes the `WebsiteContent.uuid` field name to `text_id` and changes it to a charfield, which allows us to use the config item `name` property as the id for a singleton `WebsiteContent` record
- Adds a requirement to the config schema that names must all be unique

#### How should this be manually tested?
- Add an item to "files" in the "metadata" config item for the "course" starter, then load the "Metadata" section. You should see a UI like the screenshot above
- You should be able to fill out the forms and save them. If the object did not exist in the db before, the form should be blank, and submitting should add a new object. If the object did exist, the form should be pre-populated, and submitting should update it.

You can also test that saving a site config with duplicate "name" values should cause a validation error.

#### Any background context you want to provide?
- We'll probably want a more informative UX when the user submits these forms. I'll open up an issue to implement that for these forms and any other relevant forms

#### Screenshots (if appropriate)
![ss 2021-04-16 at 22 28 45 ](https://user-images.githubusercontent.com/14932219/115099335-4964e780-9f03-11eb-91fb-9b275b9bcb22.png)
![ss 2021-04-16 at 22 29 01 ](https://user-images.githubusercontent.com/14932219/115099342-4bc74180-9f03-11eb-8634-4bd2e16bb197.png)
![ss 2021-04-09 at 13 03 09 ](https://user-images.githubusercontent.com/14932219/115099361-5681d680-9f03-11eb-9338-908b942aa56f.png)
